### PR TITLE
Checking for promises now implemented correctly for handler() method in ...

### PR DESCRIPTION
**Pull request by mkeemon**
...mixins/views/submitting and runCustomValidations in mixins/controllers/saving

### CODE REVIEW

File: addon/mixins/controllers/saving.js [hunk 1]
```diff
@@ -48,10 +48,16 @@ export default Ember.Mixin.create(
 
     /* If there is a custom validations method, resolve it */
 
-    if (runCustomValidations && !runCustomValidations.then) {
-      Ember.assert('runCustomValidations() must return a promise (e.g. return new Ember.RSVP.Promise()).');
-    } else if (runCustomValidations) {
-      runCustomValidations().then(resolve, reject);
+    if (this.runCustomValidations) {
+      customValidationsPromise = this.runCustomValidations();
```
Comment (mkeemon) [inline addon/mixins/controllers/saving.js:52]: if done the previous way (i.e.

```
var runCustomValidations = this.customValidations;
...

customValidationsPromise = runCustomValidations();
```

then  `this === window`. An alternative way to accomplish what I did here is:

```
var runCustomValidations = this.customValidations;
...

customValidationsPromise = runCustomValidations.apply(this);
```

File: addon/mixins/controllers/saving.js [hunk 2]
```diff
@@ -48,10 +48,16 @@ export default Ember.Mixin.create(
...
+
+      if (!customValidationsPromise.then) {
```
Comment (mkeemon) [inline addon/mixins/controllers/saving.js:54]: I was getting an error: `cannot read _state of undefined` when using the following:

```
Ember.assert(
  'runCustomValidations() must return a promise (e.g. return new Ember.RSVP.Promise(...)).',
  customValidationPromise.then
);
```

File: addon/mixins/views/submitting.js [hunk 3]
```diff
@@ -88,9 +88,15 @@ export default Ember.Mixin.create({
         Ember.typeOf(handler) === 'function'
       );
 
-      /* TODO - need a way to assert whether handler is a promise */
+      handlerPromise = handler();
 
-      handler().then(function() {
+      if (!handlerPromise.then) {
```
Comment (mkeemon) [inline addon/mixins/views/submitting.js:93]: Same as above

**Comment by sir-dunxalot** [comment]
Sick. Published as `0.2.8`.

**Pull request merged**